src/software/ai/hl/stp/tactic/tactic_visitor.h

```diff
@@ -145,4 +146,11 @@ class TacticVisitor
      * @param tactic The StopTestTactic to visit
      */
     virtual void visit(const StopTestTactic &tactic) = 0;
+
+    /**
+     * Visits a DefenseShadowEnemyTactic to perform an operation.
+     *
+     * @param tactic The DefenseShadowEnemyTactic to visit
+     */
+    virtual void visit(const DefenseShadowEnemyTactic &tactic) = 0;
```

Comment:
Do we have no implementations of the `TacticVisitor` that should handle this?

Reply:
Not currently AFAIK. @jonathanlew just made the interface for now.

Reply:
Ah, must be for the `MotionConstraints` stuff.